src/services/resource.js

```diff
@@ -210,6 +594,208 @@ module.exports = class resourceHelper {
 		}
 	}
 
+	static async findSequentialResources(organization_id, roles, resourceTypes = []) {
+		// get unique user roles
+		const userRoleTitles = utils.getUniqueElements(roles.map((item) => item.title))
+
+		// fetch the resource wise review levels
+		const resourceWiseLevels = await this.fetchResourceReviewLevel(organization_id, userRoleTitles, resourceTypes)
+		let resourceTypeStagesConfig = []
+		resourceTypes.filter((type) => {
+			if (resourceWiseLevels[type]) {
+				resourceTypeStagesConfig.push({
+					[Op.and]: [{ type: type }, { next_stage: resourceWiseLevels[type] }],
+				})
+			}
+		})
+		const resourceFilter = {
+			[Op.or]: resourceTypeStagesConfig,
+			organization_id,
+			status: { [Op.in]: [common.RESOURCE_STATUS_SUBMITTED, common.RESOURCE_STATUS_IN_REVIEW] },
+		}
+		const resourcesDetails = await resourceQueries.findAll(resourceFilter, ['id'])
+		let resoureId = []
+		if (resourcesDetails) {
+			resoureId = resourcesDetails.map((item) => {
+				return item.id
+			})
+		}
+		return resoureId
+	}
+
+	static async findParallelResources(organization_id, resourceTypes = []) {
+		const resourceFilter = {
+			type: {
+				[Op.in]: resourceTypes,
+			},
+			organization_id,
+			status: { [Op.in]: [common.RESOURCE_STATUS_SUBMITTED, common.RESOURCE_STATUS_IN_REVIEW] },
+		}
+		let resoureId = []
+		const resourcesDetails = await resourceQueries.findAll(resourceFilter, ['id'])
+		if (resourcesDetails) {
+			resoureId = resourcesDetails.map((item) => {
+				return item.id
+			})
+		}
+		return resoureId
+	}
+	// fetch all the resource ids from reviews table which is assigned to the reviewer
+	//  and already started reviewer
+	static async findResourcesAssignedToReviewer(organization_ids, logged_in_user_id) {
+		const reviewsFilter = {
+			organization_id: { [Op.in]: organization_ids },
+			reviewer_id: logged_in_user_id,
+			status: { [Op.in]: common.REVIEW_STATUS_UP_FOR_REVIEW },
+		}
+
+		let res = []
+
+		const reviewsResponse = await reviewsQueries.findAll(reviewsFilter, ['resource_id'])
```

Comment:
@adithyadinesh0412 The map method handles the empty array case seamlessly.
if this works then no need to check length